import_craft.py

```diff
@@ -0,0 +1,291 @@
+        ''' Rename the datas to avoid overriding while loading parts '''
+        unselect_all_objects()
+        rename_materials(ob)
+        rename_textures(ob)
+        rename_images(ob, directory)
+
+    def smooth_object_meshes(self, ob):
+        select_children(ob)
+        children = bpy.context.selected_objects
+        unselect_all_objects()
+        for child in children:
+            if child.type == "MESH":
+                bpy.context.scene.objects.active = child
+                bpy.ops.object.mode_set(mode='EDIT')
+                bpy.ops.mesh.faces_shade_smooth()
+                # Need to come back to OBJECT mode to avoid context error
+                bpy.ops.object.mode_set(mode='OBJECT')
+
+    def read_parts_models(self, partsdict, colliders):
+        for part in partsdict:
+            unselect_all_objects()
+            result = import_mu.import_mu(self, bpy.context, partsdict[part]['mu'], colliders)
+            if not result == {'FINISHED'}:
+                print('Warning : Error while importing file {}'.format(partsdict[part]['mu']))
+                return
+            partsdict[part]['object'] = bpy.context.scene.objects.active
+
+            self.smooth_object_meshes(partsdict[part]['object'])
+
+            directory = os.path.dirname(os.path.realpath(partsdict[part]['mu']))
+            self.rename_data_elements(partsdict[part]['object'], directory)
+
+    def apply_craft_transformations(self, part_node, blender_object):
+        ''' Apply .craft file transformations '''
+        blender_object.location = part_node['pos']
+        blender_object.rotation_mode = 'QUATERNION'
+        blender_object.rotation_quaternion = part_node['rot']
+
+    def generate_parts(self, parts, main, partsdict):
+        ''' Generate parts and set it according to corresponding data'''
+        for part in parts:
+            bpy.context.scene.objects.active = partsdict[part['name']]['object']
+            copyobj = partsdict[part['name']]['object']
+
+            # Duplicate the prefab and apply the transform
+            duplicated = duplicate_object_hierarchy(copyobj)
+            self.apply_craft_transformations(part, duplicated)
+            bpy.context.scene.objects.active = duplicated
+            duplicated.name = part['name'] + '_' + part['key']
+            # Pars that have a rescaleFactor parameter need to
+            # be rescaled by 1.25
+            # see http://wiki.kerbalspaceprogram.com/wiki/CFG_File_Documentation#Asset_Parameters
+            if 'rescaleFactor' not in partsdict[part['name']]:
+                duplicated.scale = (1.25, 1.25, 1.25)
+            duplicated.parent = main
+            part['object'] = duplicated
+
+        unselect_all_objects()
+
+    def read_craft_node(self, node, parts_files):
+        ''' reads a node and generate the corresponding part'''
+        part = dict()
+        # A part is PART{} with 'PART' as node[0], so we directly read node[1]
+        for value in node[1].values:
```

Comment:
probably more readable to unpack `value` with something like
`for label, value in node[1].values` as `value[0]` or `value[1]` is not readable